opba-api-security/src/main/java/de/adorsys/opba/api/security/external/mapper/HttpRequestToDataToSignMapper.java

```diff
@@ -75,4 +76,19 @@ public ConfirmConsentDataToSign mapToConfirmConsent(HttpServletRequest request,
 
         return new ConfirmConsentDataToSign(UUID.fromString(xRequestId), instant, OperationType.valueOf(operationType));
     }
+
+    public PaymentInitiationDataToSign mapToPaymentInititation(HttpServletRequest request, Instant instant) {
+        String xRequestId = request.getHeader(HttpHeaders.X_REQUEST_ID);
+        String operationType = request.getHeader(HttpHeaders.X_OPERATION_TYPE);
+
+        return new PaymentInitiationDataToSign(
```

Comment:
Looks like payment amount and currency is missing from signature - if it is the case  then we should create a bug

Reply:
Issue was created. https://github.com/adorsys/open-banking-gateway/issues/747